content/docs/rfcs/43/README.md

```diff
@@ -0,0 +1,354 @@
+and that device `A` will in any case detect a mismatch and abort the pairing, 
+regardless of the fact that the QR timeboxing mitigation is implemented or not.
+
+## Randomized Rekey
+
+The Noise Protocol framework supports [`Rekey()`](http://www.noiseprotocol.org/noise.html#rekey) 
+in order to update encryption keys *"so that a compromise of cipherstate keys will not decrypt older* \[exchanged\] *messages"*. 
+However, if a certain cipherstate key is compromised, 
+it will be possible for the attacker not only to decrypt messages encrypted under that key, 
+but also all those messages encrypted under any successive new key obtained through a call to `Rekey()`.
+
+This can be mitigated by:
+- keeping the full Handhshake State even after the handshake is complete (*by Noise specification a call to [`Split()`](http://www.noiseprotocol.org/noise.html#the-symmetricstate-object) should delete the Handshake State*)
+- continuing updating the Handshake State by processing every after-handshake exchanged message (i.e. the `payload`) according to the Noise [processing rules](http://www.noiseprotocol.org/noise.html#processing-rules) (i.e. by calling `EncryptAndHash(payload)` and `DecryptAndHash(payload)`);
+- adding to each (or every few) message exchanged in the transfer phase a random ephemeral key `e` and perform Diffie-Hellman operations with the other party's ephemeral/static keys in order to update the underlying CipherState and recover new random inbound/outbound encryption keys by calling [`Split()`](http://www.noiseprotocol.org/noise.html#the-symmetricstate-object).
+
+In short, the transfer phase would look like (but not necessarily the same as):
+
+```
+TransferPhase:
+   -> eA, eAeB, eAsB  {payload}
+   <- eB, eAeB, sAeB  {payload}
+   ...
+   
+{}: payload
+```
+
+## Messages Nametag Derivation
+
+To reduce metadata leakages and increase devices's anonymity over the p2p network, 
+[35/WAKU2-NOISE](https://rfc.vac.dev/spec/37/#session-states) suggests to use some common secrets `ctsInbound, ctsOutbound` (e.g. `ctsInbound, ctsOutbound = HKDF(h)` 
+where `h` is the [handshake hash value](https://noiseprotocol.org/noise.html#overview-of-handshake-state-machine) of the Handshake State at some point of the pairing phase) 
+in order to frequently and deterministically change the `messageNametag` of messages exchanged during the pairing and transfer phase - 
+ideally, at each message exchanged. 
+
+Given the proposed construction,
+the `ctsInbound` and `ctsOutbound` secrets can be used to iteratively generate the `messageNametag` field of Waku payloads 
+for inbound and outbound messages, respectively. 
+
+The derivation of `messageNametag` should be deterministic only for communicating devices 
+and independent from message content, 
+otherwise lost messages will prevent computing the next message nametag. 
+A possible approach consists in computing the `n`-th `messageNametag` as `H( ctsInbound || n)`, 
```

Comment:
If one wishes to use the rekeying solution sketched in the prior section, then `ctsInbound` is a function of the ephemeral key embedded in the messages, thus lost messages would cause issues in computing the next message nametag (or not?)